geopmpy/test/TestAgent.py

```diff
@@ -7,16 +7,94 @@
 
 import unittest
 import json
+import math
 from importlib import reload
 
 import geopmpy.agent
-from geopmdpy import gffi
+from . import mock_libgeopm
 
+MOCKED_AGENT_NAMES = ['agent1', 'agent2']
+MOCKED_SAMPLE_NAMES = ['sample1', 'sample2']
+MOCKED_POLICY_NAMES_1 = ['param11', 'param12']
+MOCKED_POLICY_NAMES_2 = ['param2']
+
+def mock_agent_num_avail(num_agent):
+    num_agent.__getitem__.return_value = len(MOCKED_AGENT_NAMES)
+    return 0
+
+
+def mock_agent_name(agent_idx, name_max, buff):
+    name = MOCKED_AGENT_NAMES[agent_idx]
+    for idx, char in enumerate(name):
+        buff[idx] = ord(char.encode())
+    buff[len(name)] = ord(b'\x00')
+    return 0
+
+
+def mock_agent_num_sample(agent_name, num_sample):
+    num_sample.__getitem__.return_value = len(MOCKED_SAMPLE_NAMES)
+    return 0
+
+
+def mock_agent_sample_name(agent_name, sample_idx, name_max, buff):
+    name = MOCKED_SAMPLE_NAMES[sample_idx]
+    for idx, char in enumerate(name):
+        buff[idx] = ord(char.encode())
+    buff[len(name)] = ord(b'\x00')
+    return 0
+
+
+def mock_agent_num_policy(agent_name, num_policy):
+    if agent_name.decode().find('agent1') == 0:
+        num_policy.__getitem__.return_value = len(MOCKED_POLICY_NAMES_1)
+    elif agent_name.decode().find('agent2') == 0:
+        num_policy.__getitem__.return_value = len(MOCKED_POLICY_NAMES_2)
+    else:
+        return -1
+    return 0
+
+def mock_agent_policy_name(agent_name, policy_idx, name_max, buff):
+    if agent_name.decode().find('agent1') == 0:
+        policy = MOCKED_POLICY_NAMES_1[policy_idx]
+    elif agent_name.decode().find('agent2') == 0:
+        policy = MOCKED_POLICY_NAMES_2[policy_idx]
+    assert(name_max > len(policy))
+    for idx, char in enumerate(policy):
+        buff[idx] = ord(char.encode())
+    buff[len(policy)] = ord(b'\x00')
+    return 0
+
+def mock_agent_enforce_policy():
+    return 0
+
+def mock_agent_policy_json(agent_name, policy_array, policy_max, output):
+    MOCKED_POLICY_NAMES_1 = ['param11', 'param12']
+    policy = dict()
+    if agent_name.decode().find('agent1') == 0:
```

Comment:
Maybe use `MOCKED_AGENT_NAMES[0]` instead of `'agent1'`, or do some kind of loop instead of this `if..else`